src/Contacts.js

```diff
@@ -0,0 +1,172 @@
+import { observer } from "mobx-react";
+import React from "react";
+import { action, makeObservable, observable } from "mobx";
+import { Button, Card, InputGroup } from "@blueprintjs/core";
+import { contactKey, countryKey } from "./common";
+import { addOrUpdateContact, deleteContact, getLocalData, setLocalData } from "./localCrud";
+
+const Contacts = observer( class Contacts extends React.Component {
+    emailError = false;
+    nameError = false;
+
+    constructor(props) {
+        super(props);
+        this.countryCode = this.props.searchParams.get(countryKey);
+        this.localData = getLocalData(this.countryCode)
+        this.currentContact = this.localData.contacts[this.props.index] ??
+            { name: "",
+            email: "",
+            comment: "" };
+        this.editable = this.props.index == this.props.searchParams.get(contactKey)
+
+
+        makeObservable(this, {
+            currentContact:observable,
+            countryCode:observable,
+            emailError:observable,
+            nameError:observable,

```

Comment:
this error stuff could be way cleaner if it was abstracted behind another component that does all the validation stuff for you, that way you wont have all these error tracking flags in the state of one component